src/Tools/dotnet-format/CodeFormatter.cs

```diff
@@ -94,52 +98,67 @@ private static async Task<int> FormatFilesInSolutionAsync(ILogger logger, Soluti
+                if (!workspace.TryApplyChanges(formattedSolution))
+                {
+                    logger.LogError(Resources.Failed_to_save_formatting_changes);
+                    return 1;
+                }
             }
 
             return 0;
         }
 
         private static async Task<Solution> FormatFilesInProjectAsync(ILogger logger, Project project, ICodingConventionsManager codingConventionsManager, EditorConfigOptionsApplier optionsApplier, CancellationToken cancellationToken)
         {
-            var formattedSolution = project.Solution;
-
             var isCommentTrivia = project.Language == LanguageNames.CSharp
                 ? IsCSharpCommentTrivia
                 : IsVisualBasicCommentTrivia;
 
+            var formattedDocuments = new List<(DocumentId documentId, Task<SourceText> formatTask)>();
             foreach (var documentId in project.DocumentIds)
             {
-                var document = formattedSolution.GetDocument(documentId);
+                var document = project.Solution.GetDocument(documentId);
                 if (!document.SupportsSyntaxTree)
                 {
                     continue;
                 }
 
-                var syntaxTree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
-                if (GeneratedCodeUtilities.IsGeneratedCode(syntaxTree, isCommentTrivia, cancellationToken))
+                var formatTask = Task.Run(async () =>
                 {
-                    continue;
-                }
+                    var syntaxTree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+                    if (GeneratedCodeUtilities.IsGeneratedCode(syntaxTree, isCommentTrivia, cancellationToken))
+                    {
+                        return null;
+                    }
+
+                    logger.LogTrace(Resources.Formatting_code_file_0, Path.GetFileName(document.FilePath));
+
+                    OptionSet documentOptions = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+
+                    var codingConventionsContext = await codingConventionsManager.GetConventionContextAsync(document.FilePath, cancellationToken).ConfigureAwait(false);
+                    if (codingConventionsContext?.CurrentConventions != null)
+                    {
+                        documentOptions = optionsApplier.ApplyConventions(documentOptions, codingConventionsContext.CurrentConventions, project.Language);
+                    }
 
-                logger.LogTrace(Resources.Formatting_code_file_0, Path.GetFileName(document.FilePath));
+                    var formattedDocument = await Formatter.FormatAsync(document, documentOptions, cancellationToken).ConfigureAwait(false);
+                    return await formattedDocument.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                });
```

Comment:
intentional that you're not passing cancellationToken to Task.Run?  I must admit that's one of hte least familiar Task overloads to me, so i'm not sure if this is ok or not.  Would Task.Run translate an inner cancellation exception to an aggregate exception if you don't pass cancellationToken to it?  or does it keep the OpCanceledException?

Reply:
➡️ Fixed by amending the first commit.